database/src/main/postgres/runs/V1.9.7__create_partitioning.sql

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION runs.create_partitioning(
+    IN  i_partitioning          JSONB,
+    IN  i_by_user               TEXT,
+    IN  i_parent_partitioning   JSONB = NULL,
+    OUT status                  INTEGER,
+    OUT status_text             TEXT,
+    OUT id_partitioning         BIGINT
+) RETURNS record AS
+$$
+    -------------------------------------------------------------------------------
+--
+-- Function: runs.create_partitioning(3)
+--      Creates a partitioning entry
+--
+-- Parameters:
+--      i_partitioning          - partitioning which existence to check
+--      i_by_user               - user behind the change
+--      i_parent_partitioning   - parent partitioning of the provided partitioning, optional
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_partitioning     - id of the partitioning
+--
+-- Status codes:
+--      11                  - Partitioning created
+--      12                  - Partitioning parent registered
+--      31                  - Partitioning already present
+--
+-------------------------------------------------------------------------------
+DECLARE
+    _fk_parent_partitioning BIGINT := NULL;
+    _create_partitioning    BOOLEAN;
+    _status                 BIGINT;
+BEGIN
+
+id_partitioning := runs._get_id_partitioning(i_partitioning, true);
+
+_create_partitioning := id_partitioning IS NULL;
+
+IF i_parent_partitioning IS NOT NULL THEN
+    SELECT CPINE.id_partitioning
+    FROM runs.create_partitioning_if_not_exists(i_parent_partitioning, i_by_user, NULL) AS CPINE
+    INTO _fk_parent_partitioning;
+END IF;
+
+
+IF _create_partitioning THEN
+    INSERT INTO runs.partitionings (partitioning, created_by)
+    VALUES (i_partitioning, i_by_user)
+    RETURNING partitionings.id_partitioning
+        INTO create_partitioning.id_partitioning;
+
+    PERFORM 1
+    FROM flows._create_flow(id_partitioning, i_by_user);
+
+    status := 11;
+    status_text := 'Partitioning created';
+ELSE
+    status := 31;
```

Comment:
what, is this the only difference, the status code for situation when partitioning is already present? In comparison to `runs.create_partitioning_if_not_exists` function. Otherwise I think this is 1:1 copy almost

Reply:
so I think we shouldn't have same implementation twice..possibly just call `runs.create_partitioning_if_not_exists` from this function and change the status code in select / case when statement, if this is really needed

Reply:
There is a difference in that the new function does register parent partitioning only when the child partitioning does not exist. Anyway I think we should have a call maybe about the create partitioning operation in general ...

Reply:
Let's find a spot tomorrow. I think/hope it won't be long.